studio/pages/join/index.tsx

```diff
@@ -0,0 +1,257 @@
+  const {
+    token_does_not_exist,
+    email_match,
+    authorized_user,
+    expired_token,
+    organization_name,
+    invite_id,
+  } = tokenValidationInfo || {}
+
+  const loginRedirectLink = `/?next=${encodeURIComponent(
+    `/join?token=${router.query.token}&slug=${router.query.slug}`
+  )}`
+
+  useEffect(() => {
+    const fetchTokenInfo = async () => {
+      await ui.load()
+
+      const response = await get(`${API_URL}/organizations/${slug}/members/join?token=${token}`)
+      if (response.error && response.error.code === 401) {
+        setTokenValidationInfo({
+          authorized_user: false,
+        })
+      } else {
+        setTokenInfoLoaded(true)
+        setTokenValidationInfo(response)
+      }
+    }
+
+    if (router.query.token && !tokenInfoLoaded) {
+      fetchTokenInfo()
+    }
+  }, [router.query])
+
+  async function handleJoinOrganization() {
+    setIsSubmitting(true)
+    const response = await post(`${API_URL}/organizations/${slug}/members/join?token=${token}`, {})
+    if (response.error) {
+      ui.setNotification({
+        category: 'error',
+        message: `Failed to join organization: ${response.error.message}`,
+      })
+      setIsSubmitting(false)
+    } else {
+      setIsSubmitting(false)
+      app.organizations.load()
+      router.push('/')
+    }
+  }
+
+  async function handleDeclineJoinOrganization() {
+    setIsSubmitting(true)
+    const response = await delete_(
+      `${API_URL}/organizations/${slug}/members/invite?invited_id=${invite_id}`,
+      {}
+    )
+    if (response.error) {
+      ui.setNotification({
+        category: 'error',
+        message: `Failed to decline invitation: ${response.error.message}`,
+      })
+      setIsSubmitting(false)
+    } else {
+      setIsSubmitting(false)
+      router.push('/')
+    }
+  }
+
+  const isError =
+    !!(tokenInfoLoaded && token_does_not_exist) ||
+    (tokenInfoLoaded && !email_match) ||
+    (tokenInfoLoaded && expired_token)
+
+  const ErrorMessage = () => {
+    const Container = ({ children }: { children: React.ReactNode }) => (
+      <div
+        className={[
+          'flex flex-col items-center justify-center gap-3 text-sm',
+          isError ? 'text-scale-1100' : 'text-scale-1200',
+        ].join(' ')}
+      >
+        {/* <IconAlertCircle size={21} strokeWidth={1.5} /> */}
+        <>{children}</>
+      </div>
+    )
+
+    const message = token_does_not_exist ? (
+      <>
+        <p>The invite token is invalid.</p>
+        <p className="text-scale-900">
+          Try copying and pasting the link from the invite email, or ask the organization owner to
+          invite you again.
+        </p>
+      </>
+    ) : !email_match ? (
+      <>
+        <p>
+          Your email address {profile?.primary_email} does not match the email address this
+          invitation was sent to.
+        </p>
+        <p className="text-scale-900">
+          To accept this invitation, you will need to{' '}
```

Comment:
![image](https://user-images.githubusercontent.com/19742402/174978491-aeeb91c8-0d27-48a6-b1bf-5d12deda65fd.png)

feels like we're repeating this message twice with "You will need to sign in to accept this invitation" below as well

Reply:
ah wait i landed into this state because my url was `/join?token=...` (missing slug) - though not sure if this might be something we want to catch cc @MildTomato